addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1696,6 +1696,9 @@ const Wysiwyg = Widget.extend({
                         }
                         // Unstash the mutations now that the color is picked.
                         this.odooEditor.historyUnstash();
+                        if(ev.data.target.classList.contains('o_colorpicker_reset')){
+                            ev.data.color = '';
+                        }
```

Comment:
@qsm-odoo I'm confused as to how the reset color button of the color palette is supposed to operate. From what I could gather it seems like it ends up using the color defined in `options.selectedColor`, which is `black`. But it's wrong IMO as the reset button should remove the color, not act like the black button, or any other color for that matter. This is especially obvious when resetting the background color as you end up with a back background color.

Here @maad-odoo tried a fix which consists in detecting if the color button that was clicked was in fact the _reset_ button, in which case the wrong color (black) is undone.

I think it would make more sense to not treat the reset button like a color button at all, that is remove its `o_we_color_btn` class that is used to bind the js logic, and bind a simplified custom logic to the reset button directly. What do you think ? Am I missing something here ?

Reply:
Well I disagree that it is more code because you need code to handle the special falsy color anyway. I would even argue that it would probably be easier to handle the reset separately in the code than to handle it using a weird color value that you then need to handle and convert in the rest of the code that expects an actual color value.

That being said, I agree with you that if it works properly with that system in 14.0 then it should at least also work in later version. @maad-odoo Could you investigate why it works in 14.0 but not in 16.0 please ?

Reply:
I forgot about the preview feature. Indeed my feeling of it being simpler if done separately is only true if there is no preview for the reset button. If we want the preview for the reset button then you're right that it is easier to reuse the existing mechanism.

Reply:
@dmo-odoo @qsm-odoo 
I investigated this issue again as said, and found out, that `ColorpickerUserValueWidget` is used to color the snippet background and in that case the reset-color button is working fine.

When `ColorPaletteWidget` is called, it actually sets color to the corresponding colorName.

Now incase of `ColorpickerUserValueWidget` when `ColorPaletteWidget` is called we get `#000000` value from `_getCSSVariableValue()` for black key.

But in the case for changing background or foreground of a text we get `''` value (`''` in 16.0 and `#000000` in 14.0 and 15.0) from `_getCSSVariableValue()` for black key. 

So, when we click the reset button then `''` (blank) value is passed and since black is present corresponding to `''`, then we get black color on our selected text/background. 

This the reson why I explicitly wrote that condition where it check for reset button and pass `''` as `ev.data.color` to it.

Reply:
@maad-odoo I am not sure to understand your explanation, especially since you seem to say that 14.0 and 15.0 are the same but in 14.0, there is no functional problem and in 15.0 there is.

In any case, this cannot be the right fix: here you change the code that *created* the `ColorPaletteWidget` instance to say "if the colorpalette sends me something strange, fix it". This is not a solution, I don't know what the solution is but I imagine something like:
- The widget instance is not properly initialized
- The widget code should be fixed to not send inconsistent stuff (receiving "color_picked" when clicking the reset button and having something in `ev.data.color` surely seems very wrong)... but it would be strange that this is the problem since the colorpalette reset button seems to work in all other usecases :grimacing:

Reply:
@dmo-odoo @qsm-odoo 
I looked into the points that you mentioned about the widget not being properly initialized and the other point where widget sends inconsistent stuff. I think this problem arises because we get inconsistent data from scss itself. I observed that even before any color is picked that color palette is made and color is assigned to  its corresponding colorName. And from `htmlSyle` we are getting faulty data.

I tried to find out from which color palette the color extracted from but I am unable to find one. It seems that there isn't any consistent file or palette from which all the colors are picked. But, if we declare a variable in one of the initial color palettes for example in `$o-color-palettes` then we get our desired result.

Reply:
@maad-odoo Did you compare what happens in the code in 15.0 and 16.0 ? Because in 15.0 it works correctly so it might be interesting to compare what happens in one compared to the other and detect where the behavior starts to get wrong.

Reply:
@dmo-odoo 
Yes I did compare both the version with respect to this issue. In 15.0 version we are getting correct color for `black` from `htmlStyle` but in 16.0 we are not getting correct color value from `htmlStyle`. 
I also looked into the scss file but there isn't any major changes in 15.0 and 16.0 version.

Reply:
@maad-odoo Were you able to pinpont where the correct value was coming from in 15.0 ? And where does the wrong value come from in 16.0 ?

Reply:
@dmo-odoo 
In `_getCSSVariableValue` method color values are taken from the `htmlStyle` provided to it, using `getPropertyValue`  method. And when black is passed as a key we get wrong value.

Reply:
@maad-odoo I had a look at the `_getCSSVariableValue` function. It does look at the property by looking for `--${key}` so when `key` is `'black'` it looks for `--black` but I couldn't find `--black` anywhere in the code in 15.0, even though the feature works in 15.0.

Reply:
@maad-odoo I'm confused, actually. If the returned value for `black` was correct, it would actually be _black_, right ? Therefore the background color would be black, which is the bug, initially. So why do you think receiving a "correct" value for `black` would solve this issue ?

Reply:
@dmo-odoo 
![15color](https://github.com/odoo/odoo/assets/120459796/a7f211df-f90b-4af5-8b97-f01c8100fb3d)
Above is the colorToColorNames set in 15.0 version.
![16 color](https://github.com/odoo/odoo/assets/120459796/8ee4a5d4-ad73-4c6b-87cd-4b70706e8173)
Above is the colorToColorNames set of 16.0 version.

As we can see there is value present for black in 15.0 and there `""` for black in 16.0 version. 
Now to answer your question, if we receive correct value for black then there will not be `""` beside it, instead the value of black would be something like `#000000`(like in 15.0). Then if we click on reset button and `""` is passed, then we will not get black for `""` value,in fact we will get no color for reset button, which will solve our problem.

Reply:
Resolving following PR's main thread discussion